### ISSUE TITLE
[core] Very small change allowing external rendering on-top of raylib result (currently impossible)

### ISSUE THREAD

**Issue by scottmudge** [raysan5/raylib#1216]
### Issue description

When using raylib, it is impossible to rendering anything on-top of the final raylib result render, as EndDrawing() swaps buffers before allowing the user to render anything else.

This means any other UI framework, HUDs, or other elements drawn with other OpenGL libraries are obscured.

A simple change can allow the user to pass a custom function pointer to EndDrawing(), which allows them to draw whatever they like on top of the raylib result. 

Please see this small commit on my fork as an example: https://github.com/scottmudge/raylib/commit/2e4c07c9c4e81bf2dd9fe4fb6af4e40ef0f9a51a

Is this something that can be added to the core raylib fork? 

Another user encountered this problem here (https://github.com/raysan5/raylib/issues/909), and they also had to implement a custom EndDrawing() method to get around this.

**Comment by scottmudge** [comment]
I see, thank you for the answer.

Is it then worth while adding rlgl.h and raymath.h to the standard header files which get installed? I have specified raylib as a cmake dependency in my project, but by default cmake only "installs" raylib.h to the target folder when compiling raylib. I'm assuming any prebuilt packages on *NIX systems will also only include this primary header?

It would be annoying to list raylib as a dependency, but then add a caveat saying that people need to manually add two extra header files.

**Comment by dbechrd** [comment]
I got Raylib to work alongside the DearImgui OpenGL 3.3 backend, but I also had to #include glfw.h in order to get the GLFW window handle to pass to `ImGui_ImplGlfw_InitForOpenGL` backend for rendering (the Raylib API seems to only provide the native Win32 handle):
```cpp
GLFWwindow *glfwWindow = glfwGetCurrentContext();
ImGui_ImplGlfw_InitForOpenGL(glfwWindow, true);
ImGui_ImplOpenGL3_Init("#version 130");
```
I called the new frame functions at the beginning of each frame:
```cpp
 ImGui_ImplOpenGL3_NewFrame();
ImGui_ImplGlfw_NewFrame();
ImGui::NewFrame();
```
Then I used `rlDrawRenderBatchActive()` to dump Raylib's internal buffer as Ray suggested above:
```cpp
rlDrawRenderBatchActive();
ImGui::Render();
ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
EndDrawing();
```
Finally, I cleaned up before calling Raylib's `CloseWindow` at the end of my application:
```cpp
ImGui_ImplOpenGL3_Shutdown();
ImGui_ImplGlfw_Shutdown();
ImGui::DestroyContext();
```
In addition, I had to refactor all my code so that `IsKeyPressed`, `IsKeyDown`, `IsMouseButton*` etc. are not called when DearImgui's `io.WantCaptureKeyboard` or `io.WantCaptureMouse` is true, respectively.

Oddly, I also had to comment out these three lines in the `glfw3native.h` header that I retrieved from Raylib's source directory, which were causing a redefinition error (lines 95-97 inside the `#if defined(GLFW_EXPOSE_NATIVE_WIN32)` block in my version of Raylib):
```cpp
// @raysan5: Actually, only HWND handler needs to be defined
// Including windows.h could suppose symbols re-definition issues (i.e Rectangle type)
//#include <windows.h>
//typedef void *PVOID;
//typedef PVOID HANDLE;
//typedef HANDLE HWND;
```
Presumably, something included windows.h before this file, but I couldn't find any ordering of the headers that avoided the redefinition bug. Contrary to Ray's comment, this code should `#include <windows.h>` like it was doing originally rather than declaring typedefs without checking if the types already exist, as the windows.h header has include guards that prevent redefinitions.
